toon.py

```diff
@@ -12,31 +12,49 @@
 import homeassistant.helpers.config_validation as cv
 
 # Home Assistant depends on 3rd party packages for API specific code.
-REQUIREMENTS = ['toonlib==0.4.0']
+REQUIREMENTS = ['toonlib==0.5.0']
 
 _LOGGER = logging.getLogger(__name__)
 
 DOMAIN = 'toon'
 TOON_HANDLE = 'toon_handle'
+CONF_GAS = 'gas'
+DEFAULT_GAS = True
+CONF_SOLAR = 'solar'
+DEFAULT_SOLAR = False
```

Comment:
I understand the concept but since we are doing an initial update() call upon instantiating of the toon object that already has the default values. So by default solar would be off if there are no values returned since the default setting is 0 for all values. Makes sense?